components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/ExceptionCodes.java

```diff
@@ -544,6 +544,9 @@ public enum ExceptionCodes implements ErrorHandler {
      * @param errorDescription The error description.
      */
     ExceptionCodes(long errorCode, String msg, int httpErrorCode, String errorDescription) {
+        if (httpErrorCode >= 500) {
+            this.stackTrace = true;
```

Comment:
can't we make stack trace true for specific error codes? without making globally.

Reply:
Ideally, we should be showing the stacktraces for all 500 errors. So even if we are putting them to specific error codes, we'll have to be put them into all the 500 type error codes. From this we are avoiding that and make the stack trace loggable by default for all 500 type of error codes. If the stack trace of any error code should not be logged, we can override this by using the other constructor where we can pass the stacktrace boolean.